### PR TITLE
Add ability to customize navigation announcer

### LINKED ISSUE #879: Make `svelte-announcer` text configurable
**Is your feature request related to a problem? Please describe.**
I'm building for the first time my new app with svelte-kit and I found in my bundled app text like "Navigated to ...page name..." in a div with id: `svelte-announcer`.

There is no info in docs about this.

**Describe the solution you'd like**
Understanding what we are talking about. And just in case you don't agree how to disable this mechanism.

**Additional context**
If it is about accessibility it should be possible to disable it because most of my works for example are for company devices used by specific personnel who only want maximum speed (often the hardware is very poor!). They (almost) never need "help" for visual or hearing impairments. And if and when it is needed we will provide.

### PR DESCRIPTION
**Pull request by dummdidumm**
This introduces a new `$app/a11y`, which for now only exports `setNavigationAnnouncer`, which makes i possible to customize the navigation announcer, which should also solve the internationalization-part of it. The default stays untouched - which could be changed. It also could be enhanced such that one could turn it off if one returns `null` - up for discussion.
Closes #879

### Before submitting the PR, please make sure you do the following
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts

### CODE REVIEW

File: documentation/docs/05-modules.md [hunk 1]
```diff
@@ -52,6 +52,14 @@ The stores themselves attach to the correct context at the point of subscription
 - `page` is a readable store whose value reflects the object passed to `load` functions — it contains `host`, `path`, `params` and `query`
 - `session` is a [writable store](https://svelte.dev/tutorial/writable-stores) whose initial value is whatever was returned from [`getSession`](#hooks-getsession). It can be written to, but this will _not_ cause changes to persist on the server — this is something you must implement yourself.
 
+### $app/a11y
+
+```js
+import { setNavigationAnnouncer } from '$app/a11y';
+```
+
+- `setNavigationAnnouncer(announcer)` — Set a navigation announcer for route changes. The parameter is a function that gets the `document.title` and should return a string. The innermost announcer takes precedence. This means you can for example set a generic announcer in your root `$layout`, and for specific subroutes, set a specific different announcer. If no announcer is set, a default announcer will emit `Navigated to ${title}` after each navigation.
```
Comment (dummdidumm) [inline documentation/docs/05-modules.md:61]: I'm bad at wording, so this probably needs a little cleanup.

File: packages/kit/src/core/create_app/index.js [hunk 2]
```diff
@@ -169,6 +169,9 @@ function generate_app(manifest_data, base) {
 			let navigated = false;
 			let title = null;
 
+			$: navigationAnnouncer = stores.navigationAnnouncer;
+			$: navigationAnnouncement = $navigationAnnouncer.length > 0 ? $navigationAnnouncer[$navigationAnnouncer.length - 1](title) : ('Navigated to ' + title);
```
Comment (dummdidumm) [inline packages/kit/src/core/create_app/index.js:173]: Should this by `try {} catch {}`ed?

**Comment by Conduitry** [comment]
Is this the best API for something like this? It feels a little weird to me. Could we instead have another file (similar to `src/hooks.js`) where you do `export function navigationAnnouncer() { }`, and that overrides the default one? I'm worried about this API being overly imperative rather than declarative.

**Comment by dummdidumm** [comment]
I chose to not put it in one place because when you want different announcements for different routes and you'd need to model that in one place, it could get awkward quickly, also colocation would be lost. Another thing is that internationalization is probably easier to add if you have it within your `.svelte` files because you can reuse the same mechanism.

**Comment by Conduitry** [comment]
Hm, okay. What about having it be a new `<script context='module'>` export from the .svelte file, or having it be a new thing in the response from the `load` function? I do agree that this is something that needs to be configurable, but I'm concerned about rushing into an API we'll regret, or that this will end up being one weird orphaned thing that nothing else in Kit behaves similarly to.

**Comment by dummdidumm** [comment]
Making this an export from `context="module"` feels strange to me at first, I reserved those kinds of exports to SvelteKit render options in my mind.
Returning it from the `load` function feels like mixing concerns to me. Then again, the concerns are already pretty mixed in that return object, so yeah that could be a good alternative.

Totally get your concerns about not rushing this. One half of this PR for me is to get the discussion rolling, the other half was to get to know the SvelteKit codebase a little 😄

File: documentation/docs/05-modules.md (hunk 1 above)
Comment (benmccann) [reply documentation/docs/05-modules.md:61]: the text looks pretty good to me

```suggestion
- `setNavigationAnnouncer(announcer)` — Set a navigation announcer for route changes. The parameter is a function that receives the `document.title` and should return a `string`. The innermost announcer takes precedence. This means you can for example set a generic announcer in your root `$layout`, and for specific subroutes, set a specific different announcer. If no announcer is set, a default announcer will emit `Navigated to ${title}` after each navigation.
```

File: packages/kit/test/apps/basics/src/routes/accessibility/c.svelte [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<script>
+	import { setNavigationAnnouncer } from '$app/a11y';
+
+	setNavigationAnnouncer(title => 'Seitennavigation zu '+ title);
```
Comment (benmccann) [inline packages/kit/test/apps/basics/src/routes/accessibility/c.svelte:4]:
```suggestion
	setNavigationAnnouncer(title => 'Seitennavigation zu ' + title);	
```

**Comment by benmccann** [comment]
Out of the options raised I think the current implementation or making it an `export` from `context="module"` make the most sense to me. I think I'd lean towards the `export` from `context="module"` option

**Comment by Rich-Harris** [comment]
closing in favour of #1305, per yesterday's discussion

**Pull request closed without merging**